### DISCUSSION TITLE
What is the purpose of types.tsx in navigation directory ?

### DISCUSSION THREAD

**Discussion by DavidGuillerm** [obytes/react-native-template-obytes#241] [Q&A] [answered]
Hi, 

First , thanks for this starter kit, it helps a lot.

I'm wondering what is the purpose of the types.tsx file in the navigation directory ?
I thought it was used to declared navigator stacks to be used with the _navigate_ command.
But I don't see where it is loaded... 
I test with deleting all the content, it doesn't change the compilation/behavior of the application.

Did I missed something ?

Thanks

**Comment by SihamBen** [comment, marked as answer]
Hello @DavidGuillerm,
The purpose of types.tsx is to provide a global type for the RootNavigator, this way if you use `useNavigation`, your app will ensure type safety. The type will be used as the default type.
Please refer to this part of the [docs ](https://reactnavigation.org/docs/typescript#specifying-default-types-for-usenavigation-link-ref-etc) for more details

**Comment by DavidGuillerm** [comment]
Ok, I see. Thanks
